### PR TITLE
Issue/307/nan to num

### PR DESCRIPTION
**Pull request by m-aguena**
Remove nan to num, but add safeguard on eval_sigma_critical.

### CODE REVIEW

**Review by combet — APPROVED** [review]
It looks good, thanks @m-aguena!

File: clmm/theory/parent_class.py [hunk 1]
```diff
@@ -122,6 +121,10 @@ def eval_sigma_crit(self, z_len, z_src):
         float
             Cosmology-dependent critical surface density in units of :math:`M_\odot\ Mpc^{-2}`
         """
+        if z_len<=0:
+            raise ValueError(f'Redshift for lens <= 0.')
+        if np.any(np.array(z_src)<=0):
```
Comment (pennalima) [inline clmm/theory/parent_class.py:126]: Hi @m-aguena  and @combet, 
shouldn't the safeguard be z_src <= z_lens?

Comment (combet) [reply clmm/theory/parent_class.py:126]: Hi @pennalima - we discussed that, but no, we want to keep the possibility to have z_src<z_lens as to generate mock data, it allows to contaminate the sample with foreground galaxies (which does happen in real life). When z_src<z_lens, the predictions safely go zero for shear and convergence, so no problem there.

**Review by pennalima — APPROVED** [review]

**Comment by pennalima** [comment]
@combet You can merge. :)

**Pull request merged**